src/plugins/orangehrmDashboardPlugin/Service/EmployeeTimeAtWorkService.php

```diff
@@ -0,0 +1,287 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Dashboard\Service;
+
+use DateInterval;
+use DateTime;
+use DateTimeZone;
+use Exception;
+use OrangeHRM\Core\Service\DateTimeHelperService;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Core\Traits\Service\DateTimeHelperTrait;
+use OrangeHRM\Dashboard\Dao\EmployeeTimeAtWorkDao;
+use OrangeHRM\Dashboard\Dto\TimeAtWorkLastActionDetails;
+use OrangeHRM\Entity\AttendanceRecord;
+use OrangeHRM\Time\Service\TimesheetPeriodService;
+
+class EmployeeTimeAtWorkService
+{
+    use DateTimeHelperTrait;
+    use AuthUserTrait;
+
+    public const STATE_PUNCHED_IN = 'PUNCHED IN';
+    public const STATE_PUNCHED_OUT = 'PUNCHED OUT';
+
+    /**
+     * @var EmployeeTimeAtWorkDao
+     */
+    private EmployeeTimeAtWorkDao $employeeTimeAtWorkDao;
+
+    /**
+     * @var TimesheetPeriodService
+     */
+    private TimesheetPeriodService $timesheetPeriodService;
+
+    /**
+     * @var int
+     */
+    private int $totalTimeForWeek = 0;
+
+    public function getEmployeeTimeAtWorkDao(): EmployeeTimeAtWorkDao
+    {
+        return $this->employeeTimeAtWorkDao ??= new EmployeeTimeAtWorkDao();
+    }
+
+    /**
+     * @return TimesheetPeriodService
+     */
+    private function getTimesheetPeriodService(): TimesheetPeriodService
```

Comment:
`private`, above method `public`
it's okay to keep getters public. or make all private if there is no special reason